src/Domain/DomainCreators/AlignedBlocks.hpp

```diff
@@ -0,0 +1,93 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <vector>
+
+#include "DataStructures/Index.hpp"
+#include "Domain/Domain.hpp"
+#include "Domain/DomainCreators/DomainCreator.hpp"
+#include "Options/Options.hpp"
+#include "Utilities/MakeArray.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace DomainCreators {
+
+/// \ingroup DomainCreatorsGroup
+/// \brief Create a Domain consisting of multiple aligned Blocks.
+///
+/// This is useful for setting up problems with piecewise smooth initial data.
+///
+/// \note Adaptive mesh refinement can never join blocks.
+template <size_t VolumeDim, typename TargetFrame>
+class AlignedBlocks : public DomainCreator<VolumeDim, TargetFrame> {
+ public:
+  struct BlockBounds {
+    using type = std::array<std::vector<double>, VolumeDim>;
+    static constexpr OptionString help = {
+        "Coordinates of block boundaries in each dimension."};
+  };
+
+  struct IsPeriodicIn {
+    using type = std::array<bool, VolumeDim>;
+    static constexpr OptionString help = {
+        "Whether the domain is periodic in each dimension."};
+    static type default_value() { return {{false}}; }
+  };
+
+  struct InitialRefinement {
+    using type = std::array<size_t, VolumeDim>;
+    static constexpr OptionString help = {
+        "Initial refinement level in each dimension."};
+  };
+
+  struct InitialGridPoints {
+    using type = std::array<size_t, VolumeDim>;
+    static constexpr OptionString help = {
+        "Initial number of grid points in each dimension."};
+  };
+
+  using options = tmpl::list<BlockBounds, IsPeriodicIn, InitialRefinement,
+                             InitialGridPoints>;
+
+  static constexpr OptionString help = {
+      "DomainCreator useful for problems with piecewise smooth initial data.\n"
+      "AlignedBlocks creates a regular lattice of blocks whose corners are\n"
```

Comment:
I'm going to suggest the DomainCreator be named AlignedLattice or RegularLattice since AlignedBlocks only conveys the aligned-ness of the Domain, and not the tensor product structure that Lattice conveys.